doc/developer/reference/adapter/coord_interface.md

```diff
@@ -0,0 +1,243 @@
+# Coordinator Interface
+
+This document describes the external interface that the coordinator
+presents to the protocol layers (pgwire and HTTP). In general, the
+concepts exposed by the coordinator API are close to native
+pgwire concepts (prepare, bind, execute, and so on), but there is not
+a 1-to-1 correspondence, and so this document is not simply a
+restatement of the Postgres protocol documentation.
```

Comment:
This might be a good place to link to the pgwire docs: https://www.postgresql.org/docs/15/protocol-flow.html